include/crocoddyl/multibody/contacts/contact-2d.hxx

```diff
@@ -0,0 +1,121 @@
+///////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (C) 2018-2020, LAAS-CNRS, University of Edinburgh
+// Copyright note valid unless otherwise stated in individual files.
+// All rights reserved.
+///////////////////////////////////////////////////////////////////////////////
+
+namespace crocoddyl {
+
+template <typename Scalar>
+ContactModel2DTpl<Scalar>::ContactModel2DTpl(boost::shared_ptr<StateMultibody> state, const FrameTranslation& xref,
+                                             const std::size_t& nu, const Vector2s& gains)
+    : Base(state, 2, nu), xref_(xref), gains_(gains) {}
+
+template <typename Scalar>
+ContactModel2DTpl<Scalar>::ContactModel2DTpl(boost::shared_ptr<StateMultibody> state, const FrameTranslation& xref,
+                                             const Vector2s& gains)
+    : Base(state, 2), xref_(xref), gains_(gains) {}
+
+template <typename Scalar>
+ContactModel2DTpl<Scalar>::~ContactModel2DTpl() {}
+
+template <typename Scalar>
+void ContactModel2DTpl<Scalar>::calc(const boost::shared_ptr<ContactDataAbstract>& data,
+                                     const Eigen::Ref<const VectorXs>&) {
+  Data* d = static_cast<Data*>(data.get());
+  pinocchio::updateFramePlacement(*state_->get_pinocchio().get(), *d->pinocchio, xref_.frame);
+  d->v = pinocchio::getFrameVelocity(*state_->get_pinocchio().get(), *d->pinocchio, xref_.frame);
+  d->vw = d->v.angular();
```

Comment:
Why are you saving d->vw and d->vv? I don't see you using them anywhere, and you are already saving d->v.

Reply:
What would be better is to use a `const Vector3s& vw = d->v.angular();` and so on